### ISSUE TITLE
Importing addons directly from Github is not always working

### ISSUE THREAD

**Issue by torkus** [ogri-la/strongbox#436]
It was demonstrated in [this ticket](https://github.com/ogri-la/strongbox/issues/435) that some Github addons are not being correctly detected and installed.

**Comment by torkus** [comment]
ok, a quick explanation while I'm still lucid:

1. github api logic was added when classic was brand new
2. there was no release.json, no naming conventions, lots of guessing
3. in cases where there is no release.json or hints from the catalogue we still have to do some guessing

one guess/heuristic was to classify a single unclassified addon if there was a single unused game track (game flavour, retail/classic/tbc/wotlk/etc) leftover.

I've extended this logic to say, if there is a single unclassified addon and if there are multiple unused game tracks, and 'retail' is among these unused game tracks, use retail. The addon supports *some* classic game tracks but not all.

That will handle cases like this:
![Screenshot at 2024-09-01 18-58-41](https://github.com/user-attachments/assets/8e579e53-a808-4b91-b8b1-7d2f8d4eec44)

where there is a retail, tbc, wrath and classic but no cata and the 'retail' asset doesn't have an explicit 'retail' or 'mainline' in it's name.

after my work porting the github catalogue and peeking into remote zip files I think this logic could be improved still further but I think I'll push that back into major version 8
